torch/lib/c10d/reducer.hpp

```diff
@@ -105,6 +105,19 @@ class Reducer {
   // index has been used.
   std::vector<at::Tensor> get_local_used_maps_on_device() const;
 
+  // Set logging data that can be got during DistributedDataParallel
+  // construction time.
+  void set_construction_logging_data(
```

Comment:
In order to avoid duplicates/identical data should we log only if rank == 0?

Reply:
i think it is better to log all ranks' data, especially if we want to debug desync issue, all ranks' data will be useful